alns/ALNS.py

```diff
@@ -81,7 +81,9 @@ def repair_operators(self) -> List[Tuple[str, _OperatorType]]:
         """
         return list(self._r_ops.items())
 
-    def add_destroy_operator(self, op: _OperatorType, name: str = None):
+    def add_destroy_operator(
+        self, op: _OperatorType, name: Optional[str] = None
+    ):
```

Comment:
Unrelated to this PR, but mypy has finally started complaining about this (that's why master is failing)

Reply:
We don't fix dependency versions, so I think we just got a more recent mypy that complains.

At some point we should probably explicitly test the lowest supported dependency versions in a dedicated CI run, but I haven't gotten around to that yet.